### ISSUE TITLE
Converters are not correctly generated for combinations of array and composition

### ISSUE THREAD

**Issue by jacksmith15** [jacksmith15/statham-schema#32]
Present in unreleased version.

The following will not generate models with correct constructors, and the composed schemas (inside anyOf) will have no names.
```json
{
    "type": "object",
    "properties": {
        "list_any_of": {
            "type": "array",
            "items": {
                "anyOf": [
                    {
                        "type": "object",
                        "properties": {
                            "string_prop": {"type": "string"}
                        }
                    },
                    {
                        "type": "object",
                        "properties": {
                            "integer_prop": {"type": "integer"}
                        }
                    }
                ]
            }
        }
    }
}
```

**Comment by jacksmith15** [comment]
Work to resolve is ongoing on the following branch
`feature/fix_autonaming_and_composition_in_arrays`

**Comment by jacksmith15** [comment]
Fix on #32 merged into #31
